openverse_catalog/dags/providers/factory_utils.py

```diff
@@ -5,8 +5,10 @@
 from typing import Callable, Sequence
 
 from airflow.models import TaskInstance
+from airflow.utils.dates import cron_presets
 from common.constants import MediaType
 from common.storage.media import MediaStore
+from pendulum import datetime
```

Comment:
When does the catalog repository use pendulum vs the stdlib datetime?

Reply:
Ha, that's a good question. The `pendulum.DateTime` class (which is returned from `pendulum.datetime` function) inherits from `datetime.datetime`, and Airflow uses `pendulum` internally. One could probably be substituted for the other (esp. from a functionality perspective), but the values that get handed into functions by Airflow are `pendulum.DateTime` instances[^1]. So for the purposes of typing here, it's more accurate to use `pendulum.datetime` over `datetime.datetime`.

[^1]: See the example `data_interval` template variables [here](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#variables)

Reply:
Why don't we use pendulum everywhere in this repository?

Reply:
We certainly could! `datetime.datetime` is in the standard library so that's just the one I typically reach for. There's nothing stopping us from using `pendulum` everywhere instead I don't think.

Reply:
Consistency would be nice and reduces the number of things to think about or pay attention to (this function takes a regular date time, this one takes pendulum, etc), but if it doesn't make sense to only use one or the other though, no worries.